backend/ipmi.pm

```diff
@@ -150,25 +150,35 @@ sub start_serial_grab {
     my ($self) = @_;
 
     $self->{serialpid} = fork();
+    my @cmd = ('/usr/sbin/ipmiconsole', '-h', $bmwqemu::vars{IPMI_HOSTNAME}, '-u', $bmwqemu::vars{IPMI_USER}, '-p', $bmwqemu::vars{IPMI_PASSWORD});
+
+    # zypper in dumponlyconsole, check devel:openQA for a patched freeipmi version that doesn't grab the terminal
+    push(@cmd, '--dumponly');
+
+    # our supermicro boards need workarounds to get SOL ;(
+    push(@cmd, qw(-W nochecksumcheck));
+
     if ($self->{serialpid} == 0) {
         setpgrp 0, 0;
         open(my $serial, '>',  $self->{serialfile});
         open(STDOUT,     ">&", $serial);
         open(STDERR,     ">&", $serial);
-        my @cmd = (
-            '/usr/sbin/ipmiconsole', '--serial-keepalive', '-h', $bmwqemu::vars{IPMI_HOSTNAME},
-            '-u', $bmwqemu::vars{IPMI_USER},
-            '-p', $bmwqemu::vars{IPMI_PASSWORD});
-
-        # zypper in dumponlyconsole, check devel:openQA for a patched freeipmi version that doesn't grab the terminal
-        push(@cmd, '--dumponly');
-
-        # our supermicro boards need workarounds to get SOL ;(
-        push(@cmd, qw(-W nochecksumcheck));
 
         exec(@cmd);
         die "exec failed $!";
     }
+
+    # monitor serial and restart cmd if it exited with a non-zero error code;
+    while (waitpid($self->{serialpid}, WNOHANG) > 0) {
```

Comment:
This only runs once - WNOHANG will return 0 in about 100% of all cases and so you end the loop.

If it was not, it would be a busy loop in the parent, aka the backend - blocking all tests.

This loop has to be (without busy looping) in the serial grabbing child, not the parent.